### PR TITLE
Add Audit Logs Client

### PR DESCRIPTION
**Pull request by jonatascastro12**
Add Audit Logs Client with the following functions:
- CreateEvent
- CreateExport
- GetExport

### CODE REVIEW

File: pkg/auditlogs/client_test.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+		_, err := GetExport(context.TODO(), GetExportOpts{})
+		require.Error(t, err)
+	})
+}
+
+type defaultTestHandler struct {
+	header *http.Header
+}
+
+// Unit test to hit the prod api directly. Uncomment and provide an APIKey to
```
Comment (AshGodfrey) [inline pkg/auditlogs/client_test.go:283]: Is this unit test necessary? I don't think any of our other tests have this, so I'm just wondering if something is unique here that requires this or if it was a specific ask (or if other unit tests should be improved)

File: pkg/auditlogs/client.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+// Constants that enumerate the available orders.
+const (
+	Asc  Order = "asc"
+	Desc Order = "desc"
+)
+
+// Client represents a client that performs auditlogs requests to WorkOS API.
+type Client struct {
+	// The WorkOS api key. It can be found in
```
Comment (AshGodfrey) [inline pkg/auditlogs/client.go:30]: nitpick but our similar comments are inline with links (I think there's an argument for putting the links in their own section like the go docs suggest, but right now this is all just inline)

File: pkg/auditlogs/client.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+		return CreateExportResponse{}, err
+	}
+
+	var body CreateExportResponse
+	dec := json.NewDecoder(res.Body)
+	err = dec.Decode(&body)
+	return body, err
+}
+
+// GetExport creates an export of Audit Log events. You can specify some filters.
```
Comment (AshGodfrey) [inline pkg/auditlogs/client.go:246]: Is there a spot where I'd be able to see what is filtered? (Or should this comment be eventually updated with a not yet published doc?)

File: pkg/auditlogs/client_test.go (hunk 1 above)
Comment (jonatascastro12) [reply pkg/auditlogs/client_test.go:283]: So I used this code as a reference:
https://github.com/workos/workos-go/blob/dfb61a3e3268f2e8843cfe87d30e4431b1685108/pkg/audittrail/client_test.go#L249-L269

It was actually useful when I tested live. I found a few errors and I was able to fix it.

Should we remove it?

File: pkg/auditlogs/client.go (hunk 3 above)
Comment (jonatascastro12) [reply pkg/auditlogs/client.go:246]: Oh... That's a missing copy/paste. I'll update it, but it will eventually be updated after we publish the docs. 😄

**Comment by AshGodfrey** [comment]
Just waiting on any information regarding the unit test, but otherwise looks good @jonatascastro12

File: pkg/auditlogs/client.go (hunk 2 above)
Comment (jonatascastro12) [reply pkg/auditlogs/client.go:30]: So we discussed it on Slack and I'm following the convention from other parts of the repo where there's a limit of approx. 70-80 chars for breaking the line in field comments.

**Comment by jonatascastro12** [comment]
@AshGodfrey I will remove that commented test, but in another moment I should open a discussion about end-to-end tests in the SDKs.

**Review by AshGodfrey — APPROVED** [review]

File: pkg/auditlogs/client.go [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+// Constants that enumerate the available orders.
+const (
+	Asc  Order = "asc"
+	Desc Order = "desc"
+)
+
+// Client represents a client that performs auditlogs requests to WorkOS API.
+type Client struct {
+	// The WorkOS api key. It can be found in
```
Comment (maxdeviant) [inline pkg/auditlogs/client.go:30]:
```suggestion
	// The WorkOS API key. It can be found in
```

File: pkg/auditlogs/client.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+	// https://dashboard.workos.com/api-keys.
+	APIKey string
+
+	// The http.Client that is used to post audit trail events to WorkOS. Defaults
```
Comment (maxdeviant) [inline pkg/auditlogs/client.go:34]:
```suggestion
	// The http.Client that is used to post Audit Log events to WorkOS. Defaults
```

**Pull request merged**